connexion/security/security_handler_factory.py

```diff
@@ -267,6 +267,29 @@ def wrapper(request, required_scopes):
 
         return wrapper
 
+    def verify_multiple_schemes(self, schemes):
+        """
+        Verifies multiple authentication schemes in AND fashion.
+        If any scheme fails, the entire authentication fails.
+
+        :param schemes: dict: mapping scheme_name to auth function
+        :rtype: types.FunctionType
+        """
+
+        def wrapper(request, required_scopes):
+            token_info = {}
+            for scheme_name, func in schemes.items():
+                result = func(request, required_scopes)
+                if result is self.no_value:
+                    raise OAuthProblem(
```

Comment:
If raising error here, it's possible that the next auth func doesn't get called, while that one might succeed (e.g. Oauth OR (apikey1 AND apikey2)). Think need to return self.no_value instead. Downside is that the resulting error message might not be very clear if e.g. only apikey2 is missing.